### DISCUSSION TITLE
How to assign a collection variable using portman

### DISCUSSION THREAD

**Discussion by shubhambajad** [apideck-libraries/portman#362] [Q&A] [answered]
Hi Team

i have one response in the form of array like given below,

```json
[
  {
    "id": 1,
    "date": "2021-03-24",
    "name": "h",
    "group": null,
    "status": "test",
    "type": "test",
    "start_date": "2021-03-24",
    "end_date": "2021-04-27",
    "product": "test",
    "bok": "test"
  },
  {
    "id": 2,
    "date": "2021-03-24",
    "name": "h",
    "group": null,
    "status": "test",
    "type": "test",
    "start_date": "2021-03-24",
    "end_date": "2021-04-27",
    "product": "test",
    "bok": "test"
  }
]
```

here I need to store value of id i.e. 1 in collection variable for which I need to pass syntex like jsonData[0].id
so how we can pass that because portman automatically generates code for the same and attribute which are passing get attached to it. jsondata.id.

so can you please help me on that?

**Comment by thim81** [comment, marked as answer]
Hi @shubhambajad 

Typically you would do something like this in your Portman config:

```json
{
  "version": 1,
  "assignVariables": [
    {
      "openApiOperation": "POST::*",
      "collectionVariables": [
        {
          "openApiOperationId": "get-messages",
          "collectionVariables": [
            {
              "responseBodyProp": "[0].id",
              "name": "message_id"
            }
          ]
        }
      ]
    }
  ]
}
```

Where the target is `"responseBodyProp": "[0].id"` which means take the 1st item from the array in the response body, and from the item take the property `id`.
